File: 0501-0600/0507-Perfect Number/0507-Perfect Number.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }
        
        int sum = 1;
        for (int i = 2; i * i <= num; ++i) {
            if (num % i == 0) {
                sum += i;
                if (i * i != num) {
                    sum += num / i;
                }
                
                if (sum > num) {
                    return false;
                }
            }
        }
        
        return sum == num;
    }
};

File: 0501-0600/0507-Perfect Number/0507-Perfect Number.py
class Solution:
    def checkPerfectNumber(self, num: int) -> bool:
        if num <= 1:
            return False
        sum = 1
        i = 2
        while i * i <= num:
            if num % i == 0:
                sum += i
                if i*i != num:
                    sum += num // i
                if sum > num:
                    break
            i += 1
        return sum == num
